### PR TITLE
feat: c# support

### PR DESCRIPTION
**Pull request by MrArnoldPalmer**
Adds support for `c#` documentation transpilation. Adds
`/src/docgen/transpile/csharp.ts` with formatters for classes, structs,
interfaces, and enums for `c#`.

### CODE REVIEW

File: src/docgen/transpile/csharp.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    const signatures = [`public ${name}(${params})`];
+    const invocations = [isInitializer
+      ? `new ${name}(${params});`
+      : `${type.name}.${name}(${params});`];
+
+    return {
+      name,
+      parentType: type,
+      import: this.formatImport(type),
+      parameters: callable.parameters,
```
Comment (Chriscbr) [inline src/docgen/transpile/csharp.ts:74]: Would prefer to sort the parameters so that the docs look consistent between languages.

Comment (iliapolo) [reply src/docgen/transpile/csharp.ts:74]: Yeah, all parameters should be sorted by optionality:

https://github.com/cdklabs/jsii-docgen/blob/58a61628ea9b3c951b1a88304198e5e094f5f8a8/src/docgen/transpile/java.ts#L117

File: test/docgen/view/__snapshots__/struct.test.ts.snap [hunk 2]
```diff
@@ -1,5 +1,83 @@
...
+# Initializer <a name=\\"[object Object].Initializer\\"></a>
+
+\`\`\`csharp
+using Amazon.CDK.AWS.ECR;
+
+new CfnPublicRepositoryProps {
+    repositoryCatalogData
+    repositoryName
+    repositoryPolicyText
+    tags
```
Comment (iliapolo) [inline test/docgen/view/__snapshots__/struct.test.ts.snap:19]: With all other languages we came up with a way to show the type of the property inlined here, as well as an indication of optionality, showing the required ones first. Is there something we can do here?

File: src/docgen/transpile/csharp.ts [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+      typeReference: typeRef,
+      optional: parameter.optional,
+      declaration: this.formatProperty(parameter.name, typeRef),
+    };
+  }
+
+  public property(property: reflect.Property): transpile.TranspiledProperty {
+    const typeRef = this.typeReference(property.type);
+    return {
+      name: property.name,
```
Comment (iliapolo) [inline src/docgen/transpile/csharp.ts:126]: Shouldn't this be `PascalCased`? Basing it of on https://github.com/aws-samples/aws-cdk-examples/blob/master/csharp/application-load-balancer/src/ApplicationLoadBalancer/ApplicationLoadBalancerStack.cs

File: test/docgen/view/__snapshots__/property.test.ts.snap [hunk 4]
```diff
@@ -1,5 +1,22 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`csharp snapshot 1`] = `
+" \`repositoryCatalogData\`<sup>Optional</sup> <a name=\\"Amazon.CDK.AWS.ECR.CfnPublicRepositoryProps.property.repositoryCatalogData\\"></a>
+
+\`\`\`csharp
+object repositoryCatalogData
```
Comment (iliapolo) [inline test/docgen/view/__snapshots__/property.test.ts.snap:7]: I wonder if there's something mode idiomatic here. Our AWS Dotnet docs show a getter like so:

```chsharp
public virtual string AppId { get; }
```

Is this appropriate here?

**Pull request merged**